ui/tests/integration/components/namespace-picker-test.js

```diff
@@ -10,19 +10,7 @@ import sinon from 'sinon';
 import hbs from 'htmlbars-inline-precompile';
 import Service from '@ember/service';
 import { GENERAL } from 'vault/tests/helpers/general-selectors';
-
-class AuthService extends Service {
```

Comment:
I refactored slightly to use mirage instead of manually rewriting services. This makes the test more resilient by mocking the API request instead which will subsequently test any related service logic. This makes it easier to see how the API endpoint maps to the namespace service and then finally the component. For example, the namespace service manually prepends the user's root namespace to generate the list of `accessibleNamespaces` which was lost in the original test coverage. 

While I can see arguments for "scoping component tests to their inherent logic" - Vault is a complicated product and it's very hard to gain context and develop. I think it's worth testing more complex components as close to the real-world setup as possible in order to maximize test coverage.